_studio/shared/umc/codec/h264_dec/src/umc_h264_task_supplier.cpp

```diff
@@ -2729,7 +2729,12 @@ Status TaskSupplier::ProcessFrameNumGap(H264Slice *pSlice, int32_t field, int32_
     Status umcRes = UMC_OK;
     const H264SeqParamSet* sps = pSlice->GetSeqParam();
     H264SliceHeader *sliceHeader = pSlice->GetSliceHeader();
-
+	uint32_t NumShortTermRefs, NumLongTermRefs;
+	view.GetDPBList(0)->countActiveRefs(NumShortTermRefs, NumLongTermRefs);
+	if (NumShortTermRefs > 0 || NumLongTermRefs > 0) 
+	{
+		return UMC_OK;
```

Comment:
Here u even don't try to check (see `DetectFrameNumGap`) whether we have gaps or not (so actually throw away all function's code below). The second thing, the behavior of the decoder when [SPS :: gaps_in_frame_num_value_allowed_flag] is well defined (see 8.2.5.2).

Reply:
@chuanli1 , yes, it seems the change is not according to the spec. By “if (NumShortTermRefs > 0 || NumLongTermRefs > 0)” you disable all further actions even if gaps_in_frame_num_value_allowed_flag is allowed. Which is not right. However even if input stream is corrupted we need to at least try to reach better error tolerances in the decoder.  I mean, if possible we need to try to find a WA which on the one hand doesn’t contradict the spec, and on the other hand resolves the issue.

BTW, could you check behavior with other decoders (reference JM, libav, vaapi, something else)?